### ISSUE TITLE
netvm assignment problems

### ISSUE THREAD

**Issue by Ark-of-the-copepod** [QubesOS/qubes-issues#7216]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release
Assigning (or re-assigning) netvms, by cli or gui, fails intermittently. 
In addition, was able to close a netvm that was assigned to an open Appvm, no error message.
Some netvm options are not showing in some menus.

I'm unable to determine if these are separate issues or all from the same thing, so have included all in this report.

4.1.0-rc3

### Brief summary

Assigning (or re-assigning) netvms, by cli or gui, fails intermittently. Qubes Manager will crash.
Was able to close a netvm that was assigned to an open Appvm, no error message.
Some netvm options are not available on some menus.

### Steps to reproduce

- Launch a fedora-34-dvm, disp8227, already set to get network from VPN1-vm. (If it matters, all of the templates and disposables are new on this installation).
- Attempt to change disp8227's netvm to a disposable version of VPN2-vm, disp1674

In dom0:

```
[x@dom0 Desktop]$ qvm-prefs disp8227 netvm disp1674
usage: qvm-prefs [--verbose] [--quiet] [--help] [--help-properties]
                 [--hide-default] [--get] [--set] [--default]
                 VMNAME [PROPERTY] [VALUE]
qvm-prefs: error: no such property: 'netvm'
```

Sanity check
```
[x@dom0 Desktop]$ qvm-prefs disp8227 | grep netvm
netvm               -  VPN1-vm
```

Did not work with another VPN-vm, either.

- Tried via tools in Qubes Manager. Manager crashed.

50 lines of dmesg for this is below.

- tried again with disp8227 on a different VPN-vm > same "no such property: 'netvm'" error.
- closed disp8227, closed VPN1-vm.
- launched a fresh disposable from fedora-34-dvm menu entry.
- for that new disp-vm, tried to change VPN-vm from commandline > worked perfectly.

24 hours later, closed a netvm assigned to a running AppVM
- for one already-running Appvm (this vm is recently restored from backup in 4.0)
- attempt to change networking to a disp-VPN-vm:
- somehow managed to close the net-providing VM while AppVM is still running. 
- first attempt, Qubes Manager crashed
- second attempt (Qubes Domains > 'Appvm' > Settings), reports error of "failed to access 'netvm' property" (or similar - sorry)
- third attempt succeeded. Cannot recall exactly how - I think it was Qubes Manager. May also have started the original network-vm for that Appvm. 

Lastly, note that some netvm vms are not showing in menus
- started a new disp-VPN-vm
- -> Qubes Manager
- select target AppVM > right-click menu > Networking
- new disp netvm is not listed as an option
- -> AppVM Settings > networking > new disp-netvm is available
- successfully assigned new netvm to AppVM

There may have been a 30-60s interval between me invoking the left-click and Settings panel menus.

### dmesg output
```[x@dom0 Desktop]$ sudo dmesg | tail -50
[  745.993560] xen-blkback: backend/vbd/10/51712: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[  746.215414] xen-blkback: backend/vbd/10/51728: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[  746.370170] xen-blkback: backend/vbd/10/51744: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[  746.467852] xen-blkback: backend/vbd/10/51760: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[  751.820730] pciback 0000:00:14.0: cache line size of 64 is not supported
[  751.820734] pciback 0000:00:14.0: xen_pciback: cannot enable memory-write-invalidate (-22)
[  751.870190] pciback 0000:00:1d.0: cache line size of 64 is not supported
[  751.870193] pciback 0000:00:1d.0: xen_pciback: cannot enable memory-write-invalidate (-22)
[  751.938145] pciback 0000:00:1a.0: cache line size of 64 is not supported
[  751.938149] pciback 0000:00:1a.0: xen_pciback: cannot enable memory-write-invalidate (-22)
[  759.265974] audit: type=1106 audit(1642559729.046:317): pid=9731 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_keyinit,pam_limits,pam_unix acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[  759.265980] audit: type=1104 audit(1642559729.046:318): pid=9731 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[  760.687152] input: sys-usb:  USB OPTICAL MOUSE as /devices/virtual/input/input17
[  905.485914] audit: type=1130 audit(1642559875.269:319): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-tmpfiles-clean comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[  905.485919] audit: type=1131 audit(1642559875.269:320): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-tmpfiles-clean comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[ 2232.887239] audit: type=1111 audit(1642561202.499:321): pid=9953 uid=0 auid=4294967295 ses=4294967295 msg='op=change-system-time exe="/usr/sbin/hwclock" hostname=? addr=? terminal=? res=success'
[ 4922.716576] audit: type=1100 audit(1642563892.329:322): pid=10127 uid=1000 auid=1000 ses=2 msg='op=PAM:unix_chkpwd acct="x" exe="/usr/sbin/unix_chkpwd" hostname=? addr=? terminal=? res=success'
[ 4949.062816] audit: type=1103 audit(1642563918.676:323): pid=10762 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4949.068619] audit: type=1105 audit(1642563918.681:324): pid=10762 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_keyinit,pam_limits,pam_unix acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4949.154185] audit: type=1106 audit(1642563918.765:325): pid=10762 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_keyinit,pam_limits,pam_unix acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4949.154190] audit: type=1104 audit(1642563918.765:326): pid=10762 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4949.229348] audit: type=1103 audit(1642563918.842:327): pid=10766 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4949.231841] audit: type=1105 audit(1642563918.845:328): pid=10766 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_keyinit,pam_limits,pam_unix acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4952.040930] xen-blkback: backend/vbd/12/51712: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[ 4952.191039] xen-blkback: backend/vbd/12/51728: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[ 4952.383291] xen-blkback: backend/vbd/12/51744: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[ 4952.505104] xen-blkback: backend/vbd/12/51760: using 2 queues, protocol 1 (x86_64-abi) persistent grants
[ 4958.083273] audit: type=1106 audit(1642563927.690:329): pid=10766 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_keyinit,pam_limits,pam_unix acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 4958.083279] audit: type=1104 audit(1642563927.690:330): pid=10766 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="x" exe="/usr/sbin/runuser" hostname=? addr=? terminal=? res=success'
[ 5832.799574] audit: type=1111 audit(1642564802.500:331): pid=10895 uid=0 auid=4294967295 ses=4294967295 msg='op=change-system-time exe="/usr/sbin/hwclock" hostname=? addr=? terminal=? res=success'
[ 6097.418553] audit: type=1100 audit(1642565067.119:332): pid=10925 uid=1000 auid=1000 ses=2 msg='op=PAM:unix_chkpwd acct="x" exe="/usr/sbin/unix_chkpwd" hostname=? addr=? terminal=? res=success'
[ 6843.852842] audit: type=1701 audit(1642565813.553:333): auid=1000 uid=1000 gid=1001 ses=2 pid=10973 comm="qubes-qube-mana" exe="/usr/bin/python3.8" sig=6 res=1
[ 6843.924716] audit: type=1334 audit(1642565813.625:334): prog-id=42 op=LOAD
[ 6843.925286] audit: type=1334 audit(1642565813.626:335): prog-id=43 op=LOAD
[ 6843.925389] audit: type=1334 audit(1642565813.626:336): prog-id=44 op=LOAD
[ 6843.931047] audit: type=1130 audit(1642565813.631:337): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@0-10981-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[ 6844.833448] audit: type=1131 audit(1642565814.529:338): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@0-10981-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[ 6844.850496] audit: type=1334 audit(1642565814.550:339): prog-id=44 op=UNLOAD
[ 6844.850503] audit: type=1334 audit(1642565814.550:340): prog-id=43 op=UNLOAD
[ 6844.850504] audit: type=1334 audit(1642565814.550:341): prog-id=42 op=UNLOAD
[ 6955.309425] audit: type=1101 audit(1642565925.010:342): pid=11004 uid=1000 auid=1000 ses=2 msg='op=PAM:accounting grantors=pam_unix acct="x" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 6955.310226] audit: type=1123 audit(1642565925.010:343): pid=11004 uid=1000 auid=1000 ses=2 msg='cwd="/home/x/Desktop" cmd="dmesg" exe="/usr/bin/sudo" terminal=pts/6 res=success'
[ 6955.311124] audit: type=1110 audit(1642565925.011:344): pid=11004 uid=1000 auid=1000 ses=2 msg='op=PAM:setcred grantors=pam_env,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 6955.322148] audit: type=1105 audit(1642565925.022:345): pid=11004 uid=1000 auid=1000 ses=2 msg='op=PAM:session_open grantors=pam_keyinit,pam_limits,pam_keyinit,pam_limits,pam_systemd,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 6955.342128] audit: type=1106 audit(1642565925.042:346): pid=11004 uid=1000 auid=1000 ses=2 msg='op=PAM:session_close grantors=pam_keyinit,pam_limits,pam_keyinit,pam_limits,pam_systemd,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 6955.342400] audit: type=1104 audit(1642565925.043:347): pid=11004 uid=1000 auid=1000 ses=2 msg='op=PAM:setcred grantors=pam_env,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 7063.731152] audit: type=1101 audit(1642566033.431:348): pid=11019 uid=1000 auid=1000 ses=2 msg='op=PAM:accounting grantors=pam_unix acct="x" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 7063.731157] audit: type=1123 audit(1642566033.431:349): pid=11019 uid=1000 auid=1000 ses=2 msg='cwd="/home/x/Desktop" cmd="dmesg" exe="/usr/bin/sudo" terminal=pts/6 res=success'
[ 7063.733700] audit: type=1110 audit(1642566033.434:350): pid=11019 uid=1000 auid=1000 ses=2 msg='op=PAM:setcred grantors=pam_env,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'
[ 7063.740552] audit: type=1105 audit(1642566033.441:351): pid=11019 uid=1000 auid=1000 ses=2 msg='op=PAM:session_open grantors=pam_keyinit,pam_limits,pam_keyinit,pam_limits,pam_systemd,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/6 res=success'```

**Comment by andrewdavidwong** [comment]
```
[x@dom0 Desktop]$ qvm-prefs disp8227 netvm disp1674
usage: qvm-prefs [--verbose] [--quiet] [--help] [--help-properties]
                 [--hide-default] [--get] [--set] [--default]
                 VMNAME [PROPERTY] [VALUE]
qvm-prefs: error: no such property: 'netvm'
```

You forgot `--set`.

**Comment by andrewdavidwong** [comment]
The menu stuff is mostly likely a separate issue.

**Comment by Ark-of-the-copepod** [comment]
No actually, i didn't include that in the report but i also tried --set and it made no difference. 

I should have included that, sorry. I was embarrassed.

But in 4.0, i don't use that flag. So why the difference here? And why did the GUI also refuse? And that same command worked okay later in 4.1.

On 21 January 2022 9:44:01 am AEDT, Andrew David Wong ***@***.***> wrote:
>```
***@***.*** Desktop]$ qvm-prefs disp8227 netvm disp1674
>usage: qvm-prefs [--verbose] [--quiet] [--help] [--help-properties]
>                 [--hide-default] [--get] [--set] [--default]
>                 VMNAME [PROPERTY] [VALUE]
>qvm-prefs: error: no such property: 'netvm'
>```
>
>You forgot `--set`. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/QubesOS/qubes-issues/issues/7216#issuecomment-1017991984
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by marmarek** [comment]
`--set` isn't required, should work without it. I think the issue is rather confusing error message when setting the property fails for another reason (which can be seen in `journalctl`). If this is the case, it's a duplicate of https://github.com/QubesOS/qubes-issues/issues/3528

**Comment by Ark-of-the-copepod** [comment]
Sorry for delay, I've been ill. 

"more info needed" - I reinstalled to 4.0, so I can't check it again. But I read #3528 and that doesn't seem like my experience. Perhaps you've already addressed it in 4.1.0 by now. Thanks.

**Comment by andrewdavidwong** [comment]
Closing as "cannot reproduce" (we were unable to reproduce this issue). If anyone believes this is a mistake, or if anyone can reproduce the issue, please leave a comment, and we'll be happy to reopen this. Thank you.
